[receiver/filelog] Flaky test TestReadRotatingFiles/MoveCreateSequential on Windows

**Describe the bug**
A test related to the filelogreceiver component failed, then passed later with the same code changes.

**Steps to reproduce**
https://github.com/open-telemetry/opentelemetry-collector-contrib/runs/7685135690?check_suite_focus=true

**What did you expect to see?**
Passing test.

**What did you see instead?**
```
Running target 'test' in module 'receiver/filelogreceiver' as part of group 'receiver-0'
C:/ProgramData/Chocolatey/lib/mingw/tools/install/mingw64/bin/mingw32-make.exe -C receiver/filelogreceiver test
mingw32-make[2]: Entering directory 'D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/filelogreceiver'
find: 'D:aopentelemetry-collector-contribopentelemetry-collector-contribreceiverfilelogreceiver': No such file or directory
find: 'D:aopentelemetry-collector-contribopentelemetry-collector-contribreceiverfilelogreceiver': No such file or directory
go test -race -v -timeout 300s --tags="" ./...
go: downloading github.com/observiq/nanojack v0.0.0-20201106172433-343928847ebc
=== RUN   TestDefaultConfig
--- PASS: TestDefaultConfig (0.00s)
=== RUN   TestLoadConfig
--- PASS: TestLoadConfig (0.01s)
=== RUN   TestCreateWithInvalidInputConfig
=== PAUSE TestCreateWithInvalidInputConfig
=== RUN   TestReadStaticFile
=== PAUSE TestReadStaticFile
=== RUN   TestReadRotatingFiles
=== RUN   TestReadRotatingFiles/CopyTruncateTimestamped
=== PAUSE TestReadRotatingFiles/CopyTruncateTimestamped
=== RUN   TestReadRotatingFiles/MoveCreateTimestamped
=== PAUSE TestReadRotatingFiles/MoveCreateTimestamped
=== RUN   TestReadRotatingFiles/CopyTruncateSequential
=== PAUSE TestReadRotatingFiles/CopyTruncateSequential
=== RUN   TestReadRotatingFiles/MoveCreateSequential
=== PAUSE TestReadRotatingFiles/MoveCreateSequential
=== CONT  TestReadRotatingFiles/CopyTruncateTimestamped
=== CONT  TestReadRotatingFiles/CopyTruncateSequential
=== CONT  TestReadRotatingFiles/MoveCreateTimestamped
=== CONT  TestReadRotatingFiles/MoveCreateSequential
    filelog_test.go:213: 
        	Error Trace:	D:\a\opentelemetry-collector-contrib\opentelemetry-collector-contrib\receiver\filelogreceiver\filelog_test.go:213
        	Error:      	Condition never satisfied
        	Test:       	TestReadRotatingFiles/MoveCreateSequential
        	Messages:   	expected 300 but got 292 logs
--- FAIL: TestReadRotatingFiles (0.00s)
    --- PASS: TestReadRotatingFiles/CopyTruncateSequential (0.52s)
    --- PASS: TestReadRotatingFiles/CopyTruncateTimestamped (0.53s)
    --- PASS: TestReadRotatingFiles/MoveCreateTimestamped (0.96s)
    --- FAIL: TestReadRotatingFiles/MoveCreateSequential (2.85s)
=== RUN   TestStorage
=== PAUSE TestStorage
=== CONT  TestStorage
=== CONT  TestReadStaticFile
    filelog_test.go:118: Working Directory: D:\a\opentelemetry-collector-contrib\opentelemetry-collector-contrib\receiver\filelogreceiver
=== CONT  TestCreateWithInvalidInputConfig
--- PASS: TestStorage (0.11s)
--- PASS: TestCreateWithInvalidInputConfig (0.00s)
--- PASS: TestReadStaticFile (0.23s)
FAIL
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/filelogreceiver	3.692s
FAIL
mingw32-make[2]: *** [../../Makefile.Common:56: test] Error 1
```

**What version did you use?**
latest main

--- a/pkg/stanza/fileconsumer/rotation_test.go
+++ b/pkg/stanza/fileconsumer/rotation_test.go
@@ -35,7 +35,6 @@ const windowsOS = "windows"
 func TestMultiFileRotate(t *testing.T) {
 	if runtime.GOOS == windowsOS {
 		// Windows has very poor support for moving active files, so rotation is less commonly used
-		// This may possibly be handled better in Go 1.16: https://github.com/golang/go/issues/35358
 		t.Skip()
 	}
 	t.Parallel()
@@ -94,7 +93,6 @@ func TestMultiFileRotate(t *testing.T) {
 func TestMultiFileRotateSlow(t *testing.T) {
 	if runtime.GOOS == windowsOS {
 		// Windows has very poor support for moving active files, so rotation is less commonly used
-		// This may possibly be handled better in Go 1.16: https://github.com/golang/go/issues/35358
 		t.Skip()
 	}
 
@@ -347,7 +345,6 @@ func TestRotation(t *testing.T) {
 	for _, tc := range cases {
 		if runtime.GOOS != windowsOS {
 			// Windows has very poor support for moving active files, so rotation is less commonly used
-			// This may possibly be handled better in Go 1.16: https://github.com/golang/go/issues/35358
 			t.Run(fmt.Sprintf("%s/MoveCreateTimestamped", tc.name), tc.run(tc, false, false))
 			t.Run(fmt.Sprintf("%s/MoveCreateSequential", tc.name), tc.run(tc, false, true))
 		}

--- a/receiver/filelogreceiver/filelog_test.go
+++ b/receiver/filelogreceiver/filelog_test.go
@@ -20,6 +20,7 @@ import (
 	"log"
 	"os"
 	"path/filepath"
+	"runtime"
 	"sync"
 	"testing"
 	"time"
@@ -141,20 +142,25 @@ func TestReadRotatingFiles(t *testing.T) {
 			sequential:   false,
 		},
 		{
-			name:         "MoveCreateTimestamped",
-			copyTruncate: false,
-			sequential:   false,
-		},
-		{
 			name:         "CopyTruncateSequential",
 			copyTruncate: true,
 			sequential:   true,
 		},
-		{
-			name:         "MoveCreateSequential",
-			copyTruncate: false,
-			sequential:   true,
-		},
+	}
+	if runtime.GOOS != "windows" {
+		// Windows has very poor support for moving active files, so rotation is less commonly used
+		tests = append(tests, []rotationTest{
+			{
+				name:         "MoveCreateTimestamped",
+				copyTruncate: false,
+				sequential:   false,
+			},
+			{
+				name:         "MoveCreateSequential",
+				copyTruncate: false,
+				sequential:   true,
+			},
+		}...)
 	}
 
 	for _, tc := range tests {

